### ISSUE TITLE
Fails to download ota.json 

### ISSUE THREAD

**Issue by GeoSn0w** [tihmstar/tsschecker#89]
Here is a log for you, apparently it fails to download the "ota.json" and therefore fails to also parse it.

[TSSC] opening ota.json
[DOWN] downloading file https://api.ipsw.me/v2.1/ota.json/condensed
[Error] failed to download file from=https://api.ipsw.me/v2.1/ota.json/condensed to=/tmp/ota.json CURLcode=28
[JSON] counting elements
[JSON] parsing elements
[WARNING] [TSSC] error parsing cached ota.json. Trying to redownload
[TSSC] opening ota.json
[DOWN] downloading file https://api.ipsw.me/v2.1/ota.json/condensed
[Error] failed to download file from=https://api.ipsw.me/v2.1/ota.json/condensed to=/tmp/ota.json CURLcode=28
[JSON] counting elements
[JSON] parsing elements
[Error] [TSSC] parsing ota.json failed
GeoSn0w-iMAC:~ MIKE$

**Comment by tihmstar** [comment]
if you are on windows you need to have the cacert.pem in the same directory as the tsscheccker.exe

**Comment by GeoSn0w** [comment]
I am on macOS El Capitan. For normal Restore, it fetches everything properly, the issue is OTA / Beta only.

**Comment by AngryKiller** [comment]
Same problem but with macOS Sierra
`[TSSC] opening ota.json
[JSON] counting elements
[JSON] parsing elements
[WARNING] [TSSC] error parsing cached ota.json. Trying to redownload
[TSSC] opening ota.json
[DOWN] downloading file https://api.ipsw.me/v2.1/ota.json/condensed
[Error] failed to download file from=https://api.ipsw.me/v2.1/ota.json/condensed to=/tmp/ota.json CURLcode=28
[JSON] counting elements
[JSON] parsing elements
[Error] [TSSC] parsing ota.json failed`

**Comment by jihyo-song** [comment]
Try this, works on macos/linux, windows still got error. 

Go to this [website](https://api.ipsw.me/v2.1/ota.json/) -> save as "ota.json" -> copy ota.json file to folder /tmp -> re-request ota/beta blobs with tsschecker again -> voila, ota/beta blobs successfully saved.

**Comment by macgihan** [comment]
??
please help

**Comment by macgihan** [comment]
[WARNING] [TSSC] error parsing cached ota.json. Trying to redownload
[TSSC] opening ota.json
[DOWN] downloading file https://api.ipsw.me/v2.1/ota.json/condensed
[Error] failed to download file from=https://api.ipsw.me/v2.1/ota.json/condensed to=/tmp/ota.json CURLcode=28
[JSON] counting elements
[JSON] parsing elements
[Error] [TSSC] parsing ota.json failed

**Comment by voviks21** [comment]
save firmware,json manually to User/appdata/local/temp/

**Comment by RichyElBoss** [comment]
hi guys, just tried the method you've explained (going to the link and saved the page as ota.json, then I've put the file in /tmp (it overwrote the cached ota.json, and it's double the size of the file tsschecker cached in /tmp), and then I've re-runned tsschecker, but it still give me the same error. What's the problem ? thanks. I want to save iPhone5,2 iOS 8.4.1 ota SHSH but I'm not able. I'm on OS X El Capitan, also tried this in the past on macOS 10.12.x/+ (not macOS Catalina) and then I gived up that time. But now I need that OTA SHSH blobs. Help would be very appreciate

**Comment by Valehero** [comment]
> hi guys, just tried the method you've explained (going to the link and saved the page as ota.json, then I've put the file in /tmp (it overwrote the cached ota.json, and it's double the size of the file tsschecker cached in /tmp), and then I've re-runned tsschecker, but it still give me the same error. What's the problem ? thanks. I want to save iPhone5,2 iOS 8.4.1 ota SHSH but I'm not able. I'm on OS X El Capitan, also tried this in the past on macOS 10.12.x/+ (not macOS Catalina) and then I gived up that time. But now I need that OTA SHSH blobs. Help would be very appreciate

Saving Blobs for iOS 8.4.1 (iPhone 5)??
Apple is NOT signing Blobs for this iOS. Try iOS 10.3.3 and it should work.
https://ipsw.me/iPhone5,2

**Comment by RichyElBoss** [comment]
Mmmh, that’s incorrect, iOS 8.4.1, for what I remembered till earlier this year, it’s still signed in OTA for iPhone 5 (but not also) as iPhone 5 was released with iOS 6, and if you have an iPhone 5 with iOS 6, you can’t direct upgrade to any iOS later then iOS 8.4.1; you need first to upgrade via OTA to iOS 8.4.1, and only then you can upgrade to iOS 9.3.5 (A5/A5X) or iOS 10.3.3/10.3.4 (A6/A6X, basing on iPad model if WIFI-only or WIFI+Cellular). It’s the iPhone 5c which cannot upgrade, it’s an A6, but was released along the iPhone 5s, so with iOS 7.

If something has changed in this few months (don’t think) I don’t know, but earlier this year I had very easily OTA downgraded my iPad 3 to iOS 8.4.1, due to the iOS 8.4.1 OTA still signed (repeat, if anything has not changed in this few months).

**Comment by RichyElBoss** [comment]
Ok, sorry, but instead of editing for correction it’s better to re-write the comment:

Mmmh, that’s incorrect, iOS 8.4.1, for what I remembered till earlier this year, it’s still signed in OTA for iPhone 5 (but not also, it’s signed for every 32-bit iDevice, except the iPhone 4 and earlier, Pad original, iPod touch 4th gen and earlier, and except the iPhone 5c), as iPhone 5 was released with iOS 6, and if you have an iPhone 5 with iOS 6, you can’t directly OTA upgrade to any iOS later then iOS 8.4.1; you need first to upgrade via OTA to iOS 8.4.1, and only then you can OTA upgrade to iOS 9.3.5 (A5/A5X) or iOS 10.3.3/10.3.4 (A6/A6X, basing on iPad model if WIFI-only or WIFI+Cellular). It’s the iPhone 5c which has not iOS 8.4.1 OTA signed, because even if it’s an A6, it was released along the iPhone 5s, so with iOS 7.

If something has changed in this few months (don’t think) I don’t know, but earlier this year I had very easily OTA downgraded my iPad 3 to iOS 8.4.1, due to the iOS 8.4.1 OTA still signed (repeat, if anything has not changed in this few months).

Basically, it’s the same as iOS 6.1.3 still OTA signed for iPhone 4S (and some of the earlier 4s even if they never were released with iOS 5) iPad 2 (expect iPad 2 Mid 2012, iPad2,4), as if there still is an iPhone 4S with iOS 5 and you want to OTA upgrade it to iOS 9.3.6, you first need to OTA upgrade to iOS 6.1.3, and then OTA upgrade to iOS 8.4.1... and then finally OTA upgrade to iOS 9.3.6. :)  even if now iOS 6.1.3 is signed even non OTA

(Hooping I don’t need to make any other edit for correct something 😂)
